table_connector.py

```diff
@@ -86,9 +86,10 @@ def __get_address_joined_with_message_id(address_book):
     key_fields = ['message_id', 'chat_id', 'phone_or_email']
     duplicates = address_joined_with_message_id.duplicated(subset=key_fields,
                                                            keep=False)
-    error_message = ("message_id, chat_id and phone_or_email "
+    error_message = ("WARNING: tuple (message_id, chat_id, and phone_or_email) "
                      "do not form a composite key")
```

Comment:
Change this to use single quotes to be consistent with the file, and use string formatting to print out the number of duplicates.  I'm interested to hear how many occurred in your dataset.  It should be very few, otherwise there's some other error I need to debug.